modin/pandas/test/dataframe/test_indexing.py

```diff
@@ -411,6 +411,48 @@ def test_loc_assignment(index, columns):
     df_equals(md_df, pd_df)
 
 
+@pytest.fixture
+def loc_iter_dfs():
+    columns = ["col1", "col2", "col3"]
+    index = ["row1", "row2", "row3"]
+    return create_test_dfs(
+        {col: ([idx] * len(index)) for idx, col in enumerate(columns)},
+        columns=columns,
+        index=index,
+    )
+
+
+@pytest.mark.parametrize("reverse_order", [False, True])
+@pytest.mark.parametrize("axis", [0, 1])
+def test_loc_iter_assignment(loc_iter_dfs, reverse_order, axis):
+    if reverse_order and axis:
+        pytest.xfail(
+            "Due to internal sorting of lookup values assignment order is lost, see GH-#2552"
+        )
+
+    md_df, pd_df = loc_iter_dfs
+
+    select = [slice(None), slice(None)]
+    select[axis] = sorted(pd_df.axes[axis][:-1], reverse=reverse_order)
+    select = tuple(select)
+
+    pd_df.loc[select] = pd_df.loc[select].copy()
+    md_df.loc[select] = md_df.loc[select].copy()
```

Comment:
it seems that if 'loc' assignment doesn't work we will not see it in this test,
maybe we should do some modification for the right value to be sure that assignment is done? like:
```python
pd_df.loc[select] = pd_df.loc[select] + pd_df.loc[select]
md_df.loc[select] = md_df.loc[select] + md_df.loc[select]
```

Reply:
Makes sense, added